### PR TITLE
7661 tag aggregate score

### PR DESCRIPTION
**Pull request by gdicristofaro**

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java [hunk 1]
```diff
@@ -274,6 +283,49 @@ Score updateAggregateScoreAfterDeletion(long objId, Long dataSourceObjectId, Cas
...
+	 * object id.
+	 *
+	 * @param objectId   The object id of the item.
+	 * @param transaction The case db transaction to perform this query.
+	 *
+	 * @return The maximum FileKnown status for this object or empty.
+	 *
+	 * @throws TskCoreException
+	 */
+	private Optional<TskData.FileKnown> getTagKnownStatus(long objectId, CaseDbTransaction transaction) throws TskCoreException {
```
Comment (bcarrier) [inline bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java:298]: I'd move this to TaggingManager and give it name of something like getMaxTagKnownStatus().

File: bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java [hunk 2]
```diff
@@ -32,6 +32,10 @@
  * Provides an API to manage Tags.
  */
 public class TaggingManager {
+	// Score for tags with known bad status
+	private static Score BAD_SCORE = new Score(Score.Significance.NOTABLE, Score.MethodCategory.AUTO);
+	// Score for any other type of tag known status
+	private static Score UNKNOWN_SCORE = new Score(Score.Significance.LIKELY_NOTABLE, Score.MethodCategory.AUTO);
```
Comment (bcarrier) [inline bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java:38]: This is already defined in Score or ScoringManager.    We should add the BAD_SCORE to there as well (and call it NOTABLE_SCORE)

File: bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java (hunk 1 above)
Comment (gdicristofaro) [reply bindings/java/src/org/sleuthkit/datamodel/ScoringManager.java]: Sounds good.

File: bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java (hunk 2 above)
Comment (gdicristofaro) [reply bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java:39]: Hi @bcarrier ,
Are you saying you want me to use the constants I will be adding to `Score`?

Comment (bcarrier) [reply bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java:39]: Yea, use the one already in Score and add the other there. Note that this PR will have conflicts with #2431, once that gets merged into develop.

Comment (gdicristofaro) [reply bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java]: Ok.  Will do.

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 3]
```diff
@@ -11155,16 +11155,31 @@ public ContentTag addContentTag(Content content, TagName tagName, String comment
 	 * @throws TskCoreException
 	 */
 	public void deleteContentTag(ContentTag tag) throws TskCoreException {
+		CaseDbTransaction trans = beginTransaction();
 		acquireSingleUserCaseWriteLock();
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:11159]: No need to get write lock with open transaction

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java [hunk 4]
```diff
@@ -11512,15 +11527,30 @@ public BlackboardArtifactTag addBlackboardArtifactTag(BlackboardArtifact artifac
 	 */
 	public void deleteBlackboardArtifactTag(BlackboardArtifactTag tag) throws TskCoreException {
 		acquireSingleUserCaseWriteLock();
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java:11529]: Again, lock is unnecessary

File: bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java [hunk 5]
```diff
@@ -294,6 +298,66 @@ public BlackboardArtifactTagChange addArtifactTag(BlackboardArtifact artifact, T
...
+				+ "	    UNION\n"
+				+ "	    SELECT btags.tag_name_id AS tag_name_id FROM blackboard_artifact_tags btags \n"
+				+ "	    INNER JOIN blackboard_artifacts ba ON btags.artifact_id = ba.artifact_id\n"
+				+ "	    WHERE ba.artifact_obj_id = " + objectId + "\n"
+				+ "	) tag_name_ids\n"
+				+ "	INNER JOIN tag_names ON tag_name_ids.tag_name_id = tag_names.tag_name_id\n"
+				+ "	ORDER BY tag_names.knownStatus DESC\n"
+				+ "	LIMIT 1";
+
+		skCase.acquireSingleUserCaseReadLock();
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/TaggingManager.java:345]: Another unnecessary lock

File: bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java (hunk 3 above)
Comment (gdicristofaro) [reply bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java]: Ok.  Should be good to go now.

**Review by APriestman — APPROVED** [review]

**Pull request merged**